### PR TITLE
Use prettier to format the documentation

### PR DESCRIPTION
**Pull request by konstin**
To enforce the 100 character line limit in markdown files introduced in https://github.com/astral-sh/uv/pull/5635, and to automate the formatting of markdown files, i've added prettier and formatted our markdown files with it.

I've excluded the changelog and the generated references documentation from this for having too many changes, but we can also include them.

I'm not particular on which style we use. My main motivations are (major) not having to reflow markdown files myself anymore and (minor) consistence between all markdown files. I've chosen prettier for similar reason as we chose black, it's a single good style that's automated and shared in the community. I do prefer prettier's style of not breaking inside of a link name though.

This PR is in two parts, the first adds prettier to CI and documents using it, while the second actually formats the docs. When merge conflicts arise, we can drop the last commit and regenerate it with `npx prettier --prose-wrap always --write BENCHMARKS.md CONTRIBUTING.md README.md STYLE.md docs/*.md docs/concepts/**/*.md docs/guides/**/*.md docs/pip/**/*.md`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -81,6 +81,15 @@ jobs:
...
         run: pipx run --python 3.12 mypy
 
+  lint-other:
+    name: "Lint other"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Check documentation formatting"
+        run: npx prettier --prose-wrap always --check docs/*.md docs/concepts/**/*.md docs/guides/**/*.md docs/pip/**/*.md
```
Comment (zanieb) [inline .github/workflows/ci.yml:91]: Why not `docs/**/*.md`? Because we need to exclude generated files? This seems brittle.

Comment (konstin) [reply .github/workflows/ci.yml:91]: Yeah, i went with the conservative option here. I have touched neither the reference docs nor the changelog, so i'll follow your lead for them.

**Comment by konstin** [comment]
For the jetbrains IDE users, you can configure prettier to run for `*.md` on save instead of the default formatter

image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zanieb) [reply .github/workflows/ci.yml:91]: Does it support exclusions? Can we encode this in a repo-level config file?

Comment (konstin) [reply .github/workflows/ci.yml:91]: I added exclusion comments. If we have more setting prettier we can switch to one of their configuration files.

I'd like to format the generated files too, but the diff is already large enough.

**Review by BurntSushi — APPROVED** [review]
SGTM

**Pull request merged**